tests/unit/test_cli.py

```diff
@@ -193,6 +193,15 @@ def test_with_releases():
     assert str(conf.options.bad_date) == releases_data[-1][1]
 
 
+@pytest.mark.parametrize('arg', [
+    '--launch=34', '--launch=reded3rt4'
+])
+def test_launch_with_release_number(arg):
+    config = do_cli(arg)
+    assert config.action == 'launch_nightlies'
+    assert config.options.launch == cli.parse_date(releases()[34])
+
```

Comment:
This test seems broken - if you pass '--launch=reded3rt4', config.options.launch should NOT be equal to cli.parse_date(releases()[34]).

I think you can remove the parametrize here. Just test for a right hardcoded value ('--launch=34'). the two asserts are fine though :)

I don't think you should test for a bad release number (like '--launch=reded3rt4') since it is equivalent to testing a good changeset (and that you tested it in another test). :) but if you want to do that, write the test in a new function test_launch_with_bad_release_number.